packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -356,6 +349,13 @@ function DashboardLayout(props: DashboardLayoutProps) {
     }
   }, []);
 
+  // If useEffect was used, the reset would also happen on the client render after SSR which we don't need
+  React.useMemo(() => {
+    uniqueItemPaths = new Set();
```

Comment:
I don't think this can be safely done with concurrent mode. a render can be interrupted by a higher priority one and reset these.

Perhaps you can store the set in a ref and pass that as a prop to child components? then reset the ref in an effect.

Reply:
Refs work fine, I just wasn't sure about adding props just for this, but yeah global variables can fail in some rare cases so this is definitely better.

Changes here: https://github.com/mui/mui-toolpad/pull/3838/commits/c9269c6f81b377912c83f0f4e5bf9af5b6a37e53

Reply:
> (Neither would it work correctly with the theoretical case of having this component twice in the tree).

Actually have to come back on this, it's not theoretical, we are rendering this component multiple times in a page on the docs 😄

Reply:
I'll merge in a bit if this current approach with props seems good.